python/lsst/analysis/tools/tasks/makeMetricTable.py

```diff
@@ -0,0 +1,203 @@
+# This file is part of analysis_tools.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+__all__ = (
+    "MakeMetricTableConfig",
+    "MakeMetricTableTask",
```

Comment:
Suggest explicitly adding tract to these names, e.g., `MakeMetricTableTractTask`

Reply:
Or maybe `WholeSkyMetricTableTractTask` ? You call it `wholeSkyMetricTableOTC` in the pipeline

Reply:
Yeah, what _is_ OTC?

Reply:
I haven't added tract into the names yet because without doing the visit bit I am not sure shich bits are going to be tract level specific, I am hoping to also include which tract each visit overlaps, not sure how I am dealing with multiples yet but I think almost all of this code is applicable to both, except I still need to understand the AP data structures.

Reply:
I see the name change in the pipeline file, and in light of your intentions to add visit level support I think this can be resolved. If you want to be even more explicit you might add something like "TODO: Add visit level support on DM-XXXXX" in the docstring.

Reply:
I see the name change in the pipeline file, and in light of your intentions to add visit level support I think this can be resolved. If you want to be even more explicit you might add something like "TODO: Add visit level support on DM-XXXXX" in the docstring.